### PR TITLE
Contractor and Ninja Modsuits have a deployable microbomb (Enable the DNA lock and hit Lock on the modsuit to arm)

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

Alternatives to blowing up. You can drain the cells with EMP, you and acid them off... but you gotta work for it. 

## Why It's Good For The Game

They're not supposed to be loot bags. 

## Changelog

:cl:
balance: contractor and ninja modsuits have DNA locked microbombs
/:cl:

### CODE REVIEW

**Comment by Cyprex** [comment]
I really don't like this, for starters it prevents a lot of opportunities for ERP, upgrading armor, customizing your looks. You can't do stealth as a contractor by ditching your armor either- which is something I've seen some players do!

We should just revert 13694ca instead

**Comment by GoreDem** [comment]
> I really don't like this, for starters it prevents a lot of opportunities for ERP, upgrading armor, customizing your looks. You can't do stealth as a contractor by ditching your armor either- which is something I've seen some players do!
> 
> We should just revert [13694ca](https://github.com/Bubberstation/Bubberstation/commit/13694ca6e89c46d2c3ed2d6e865e8bb687267e30) instead

Contractor modsuits have the chameleon module, effectively giving stealth when its not enabled just fyi

**Comment by StrangeWeirdKitten** [comment]
Make it a module that only gives the suit No Drop when it's powered or just give the explosive implant back. If you don't want people getting RRed, then I don't see how this will change anything since they will be justified to simply not revive the ninja / contractor.

**Comment by Tupinambis** [comment]
This enforces lrp kos mentality, since it becomes entirely impractical to contain ninjas and contractors under any circumstances.

Microbomb implant at least gave contractors and ninjas the option to let themselves be captured or disarmed for rp purposes. With this pr they can't even do that.

Also some people do like stripping themselves for full stealth, and having that be an option is genuinely compelling and should be kept in the game.

**Comment by nevimer** [comment]
> This enforces lrp kos mentality, since it becomes entirely impractical to contain ninjas and contractors under any circumstances.
> 
> Microbomb implant at least gave contractors and ninjas the option to let themselves be captured or disarmed for rp purposes. With this pr they can't even do that.
> 
> Also some people do like stripping themselves for full stealth, and having that be an option is genuinely compelling and should be kept in the game.

sure they can, though security will kill any antag regardless right now.

**Comment by destrucktoid** [comment]
> > This enforces lrp kos mentality, since it becomes entirely impractical to contain ninjas and contractors under any circumstances.
> > Microbomb implant at least gave contractors and ninjas the option to let themselves be captured or disarmed for rp purposes. With this pr they can't even do that.
> > Also some people do like stripping themselves for full stealth, and having that be an option is genuinely compelling and should be kept in the game.
> 
> sure they can, though security will kill any antag regardless right now.

An issue which this PR exacerbates

**Comment by YakumoChen** [comment]
I don't see a problem with this. Handcuffs already disable almost all the threat of both antagonists, their modsuits shouldn't be removable in the first place.

It doesn't actually matter in the case of Ninja since the modsuit will just gib anyone else who puts it on regardless.

**Comment by destrucktoid** [comment]
You know what the solution to the whole lootbox debacle is? Don't give them gear you don't want to let the station have if they beat the threat. At the end of the day, the loot from an antag is the station's reward for killing them. The Space Ninja doesn't really have a problem with being a lootbox since the only thing they drop is their sword. Contractors are the real lootbox issue here and i have a few suggestions, firstly make the contractor modsuit have an acid module, this module makes it so that should someone other than the contractor open it, the items inside are coated with acid and destroyed (and the modsuit module as well), if the contractor loses their modsuit they are either dead or arrested anyway so this won't cause them issues. The big problem here is that contractor being a lootbox isn't really an issue, sec aren't allowed to use their gear and we don't have validhunters on this server so no need to worry about them dying to crewmember and being lootboxes that way.

As a result of my previous points I personally see this as a PR addressing an issue that simply doesn't exist or does not appear often enough to justify such a wide sweeping change and it is therefore unnecessary. 
My suggestions however would be the following: 
Give back explosive implants but make the explosive implant not automatically activate on death, force the person who has it to use it, thus removing the RR problems. 
Revert this PR and leave contractor as it is, there is no lootbox problem due to the nature of this server but if you REALLY want to stop people looting contractors then add a single use acid module that destroys the contents of the contractor modsuit (but not the modsuit itself).

**Comment by YakumoChen** [comment]
> , firstly make the contractor modsuit have an acid module

Code it yourself then?

**Comment by Daved27hundred** [comment]
instead of an explosion, make them fry their internal parts when they detect someone other than the user trying to operate them, making them inoperable without putting a hole in the station.

**Comment by projectkepler-ru** [comment]
Hard to say how I feel on this, I don't mind having non-removable contractor modsuit as I don't believe their item should be usable

but I also don't agree with @destrucktoid  idea of balance being that if something isn't intended they should be coded out.

As not everything in the game can ever be accounted for and coded out until an issue happen

As for the whole blowing up thing, on skyrat I've never actually seen a contractor blow up as of late even when they're killed but I haven't gone around checking why's that..  Could be worth looking into?

**Comment by YakumoChen** [comment]
> Hard to say how I feel on this, I don't mind having non-removable contractor modsuit as I don't believe their item should be usable
> 
> but I also don't agree with @destrucktoid idea of balance being that if something isn't intended they should be coded out.
> 
> As not everything in the game can ever be accounted for and coded out until an issue happen
> 
> As for the whole blowing up thing, on skyrat I've never actually seen a contractor blow up as of late even when they're killed but I haven't gone around checking why's that.. Could be worth looking into?

They removed the bomb implants from contractor and ninja

**Comment by destrucktoid** [comment]
> Hard to say how I feel on this, I don't mind having non-removable contractor modsuit as I don't believe their item should be usable
> 
> but I also don't agree with @destrucktoid idea of balance being that if something isn't intended they should be coded out.
> 
> As not everything in the game can ever be accounted for and coded out until an issue happen
> 
> As for the whole blowing up thing, on skyrat I've never actually seen a contractor blow up as of late even when they're killed but I haven't gone around checking why's that.. Could be worth looking into?

I literally never said that, do not put words into my mouth that are not true. What i said is that if you do not want the station to get something from an antagonist, do not give it to that antagonist. Which is NOT the same as saying "remove everything unintentional".

File: modular_zubbers/code/modules/antagonists/modglue.dm
Comment (BurgerLUA) [inline modular_zubbers/code/modules/antagonists/modglue.dm:4]: I don't think ``I`` needs to be casted here, and ``I`` can just be ``src`` in the ADD_TRAIT define

**Review by BurgerLUA — APPROVED** [review]

**Comment by projectkepler-ru** [comment]
> > Hard to say how I feel on this, I don't mind having non-removable contractor modsuit as I don't believe their item should be usable
> > but I also don't agree with @destrucktoid idea of balance being that if something isn't intended they should be coded out.
> > As not everything in the game can ever be accounted for and coded out until an issue happen
> > As for the whole blowing up thing, on skyrat I've never actually seen a contractor blow up as of late even when they're killed but I haven't gone around checking why's that.. Could be worth looking into?
> 
> They removed the bomb implants from contractor and ninja

Yeah that explains it

**Comment by nevimer** [comment]
> dogshit pr you need to remove the modsuits to change your equipment as well as contractor why not just give them explosive implants again

because the person who removed them on sr had good justification in my opinion

**Comment by StrangeWeirdKitten** [comment]
It's alright to dislike a PR and provide feedback, but the non-constructive comments are going to get deleted. This will have to be discussed by the staff team before it's fully merged.

**Comment by radarseas** [comment]
okay so, from two sides of this... for ninjas... ehh they'll usually be executed for their crimes of fucking up **everything** so honestly, what happens there, happens. I won't cover ninjas on feedback here. another can of worms.

**_security_**
as security, dealing with contractors with this is PAINFUL, usually you're juggling like 5-8 threats easily on highpop, let alone if most of them are going hard, contractors are, really, really frustrating to deal with, why you might ask? well they tend to have radio jammers in their modsuits that can be on indefinitely, there's **literally** no quick way to get it out of the contractor's modsuit.

contractor mods also have a baton holster (which is superior to antidrop so you can almost guarentee you're better off just straight executing them for existing regardless of what they did if they're hostile. because frankly, melting it off with acid doesn't work, its a backpack and if it does... that relies on somehow getting acid from chemistry while dealing with a LOT of other things going on, which makes it not even remotely worth the time to nonlethally deal with them.

_**contractors**_
okay, so despite being not looted, this PR really, really, **really** makes your life hell as a contractor. its nearly impossible to pass off your modsuit as "just a bag" because its almost impossible to not meta it from being unremovable, which straight up outs a contractor from a random lone infiltrator or traitor with a cham module'd suit.

you can't upgrade for an elite modsuit once you start to get a lot of telecrystals, so you're perma-stuck with bloodred values of armoring, this isn't usually an _issue, at all_ considering contractors aren't really supposed to become crew-wide threats. however it can be a real pain in the ass.

security will nearly always want to kill you for being uncontainable as high heck, oh and if your MOD does get destroyed! guess what! all of your hard earned goods and equipment is for waste, that's right baby, if your tablet was in your suit _you ain't doing your entire gimmic bucko!_ effectively de-antagging you past escalation lee-way... oh and to remove suits, usually a gib mix or bodyswap is like the only feasable way to actually deal with contractors in a timely manner if you're not shoving them in a tray for being uncontainable (unremovable antidrop two hit baton that even takes out borgs + flash resistance and armoring from your modsuit).

this also _means, most of the time, your implants including your means of progression you can't easily replace yourself_ **your uplink implant** is almost guarrented to be gone. that's right baby! you can't do anything once caught if they do decide to make you live.. you're worse off than just losing gear to crew. (which to be honest if you mess up with a lotta TC, crew kinda deserve your gear, they can get it from random supply pods anyway. (also skyrat axed all contractor telecrystals so urm, that's if you even get to getting TC.)

crew like medbay are just gonna throw you into the morgue or a plasma lake like, all of the time, why would you even bother handing security a headache if they can't even contain said contractor.. its kind of justified to do so and I've had it happen to me, knocking me out of a round.

there's other reasonings you might want to remove the suit, escaping searches, picking a crew suit to sneak into a department or around while security assumes you're wearing your fancy modsuit while laying low post-objectives, ERP scenes,  going hardmode contractor, etc.


as for crew stealing the suit and wearing it, well... they're subject to whatever's thrown at them for dressing like a murderious contractor agent and honestly, _act/look like an antag, be treated like one._ they can already do it with bloodred/elite MODs so there's really little difference apart from conceilablity (less movespeed difference + cham module) from a contractor MOD and that can be countered with xenobio on the rare case its actually done in a round via speed potions.

I think contractor MODs should be allowed to be removed, just because it causes the antagonist and security more unnessary problems than other antags from them losing their gear.. to begin with and likely ninjas, contractors are REALLY, common midrounds.

**Comment by nevimer** [comment]
> okay so, from three (technically four?) sides of this... for ninjas... ehh they'll usually be executed for their crimes of fucking up **everything** so honestly, what happens there, happens. I won't cover ninjas on feedback here. another can of worms.
> 
> **_security_** as security, dealing with contractors with this is PAINFUL, usually you're juggling like 5-8 threats easily on highpop, let alone if most of them are going hard, contractors are, really, really frustrating to deal with, why you might ask? well they tend to have radio jammers, stolen important shit (like the spare ID or high risk items), there's **literally** no quick way to get it out of the contractor's modsuit.
> 
> contractor mods also have a baton holster (which is superior to antidrop so you can almost guarentee you're better off just straight executing them for existing regardless of what they did if they're hostile. because frankly, melting it off with acid doesn't work, its a backpack and if it does... that relies on somehow getting acid from chemistry while dealing with a LOT of other things going on, which makes it not even remotely worth the time to nonlethally deal with them.
> 
> not to mention its a full suit of armor that's spaceworthy with means to break every window you see and you can't remove it, you're dealing with a threat that has the escapability of a changling unwatched and absolutely none of the benefits to actually let it do that without dying... it just causes sec to have to babysit a contractor until they somehow get the MOD off or kill them. every, _single_, time. there's no means of RP with them past killing them and absolutely zero come back, they can't even raid security for their tablet again if its in their MOD.
> 
> a lot of security splash acid on the ground which... doesn't actually melt the modsuits like its supposed to, (and if it did you could theoretically wipe out a contractor's entire set of gear and equipment, **instantly** without any means whatsoever of recovery.
> 
> _**contractors**_ okay, so despite being not looted, this PR really, really, **really** makes your life hell as a contractor. its nearly impossible to pass off your modsuit as "just a bag" because its almost impossible to not meta it from being unremovable, which straight up outs a contractor from a random lone infiltrator or traitor with a cham module'd suit.
> 
> you can't upgrade for an elite modsuit once you start to get a lot of telecrystals, so you're perma-stuck with bloodred values of armoring, this isn't usually an _issue, at all_ considering contractors aren't really supposed to become crew-wide threats. however it can be a real pain in the ass.
> 
> security will nearly always want to kill you for being uncontainable as high heck, oh and if your MOD does get destroyed! guess what! all of your hard earned goods and equipment is for waste, that's right baby, if your tablet was in your suit _you ain't doing your entire gimmic bucko!_ effectively de-antagging you past escalation lee-way... oh and to remove suits, usually a gib mix or bodyswap is like the only feasable way to actually deal with contractors in a timely manner if you're not shoving them in a tray for being uncontainable (unremovable antidrop two hit baton that even takes out borgs + flash resistance and armoring from your modsuit).
> 
> this also _means, most of the time, your implants including your means of progression you can't easily replace yourself_ **your uplink implant** is almost guarrented to be gone. that's right baby! you can't do anything once caught if they do decide to make you live.. you're worse off than just losing gear to crew. (which to be honest if you mess up with a lotta TC, crew kinda deserve your gear, they can get it from random supply pods anyway. (also skyrat axed all contractor telecrystals so urm, that's if you even get to getting TC.)
> 
> crew like medbay are just gonna throw you into the morgue or a plasma lake like, all of the time, why would you even bother handing security a headache if they can't even contain said contractor.. its kind of justified to do so and I've had it happen to me, knocking me out of a round.
> 
> there's other reasonings you might want to remove the suit, escaping searches, picking a crew suit to sneak into a department or around while security assumes you're wearing your fancy modsuit while laying low post-objectives, ERP scenes, going hardmode contractor, etc.
> 
> as for crew stealing the suit and wearing it, well... they're subject to whatever's thrown at them for dressing like a murderious contractor agent and honestly, _act/look like an antag, be treated like one._ they can already do it with bloodred/elite MODs so there's really little difference apart from conceilablity (less movespeed difference + cham module) from a contractor MOD and that can be countered with xenobio on the rare case its actually done in a round via speed potions.
> 
> I think contractor MODs should be allowed to be removed, just because it causes the antagonist and security more unnessary problems than other antags, without even having much more benefit than dying to a random assistant that sec doesn't confiscate contraband from a combat suit (which honestly something almost on-par be easily made with retracting plates from cargo) and some various B&E tools (you can get these from cargo, **plus a drill that for some forsaken reason fits in a pocket and does three tools in one, ah yes thanks skyrat**) a MOD tied baton (to be honest you can get antidrop implants anyway if you want to reach something on-par) and uhh... maybe a weapon if the contractor got that far to begin with and unlike _space ninjas_, contractors are **REALLY**, common midrounds.
> 
> _**crew/antags**_ you can't really confiscate a radio jammer off of them which makes calling for sec to take their dead body an issue. oh and if you're the chaplain and soul shard them... well urm, _you kinda have unremovable syndicate contraband_... that security can't even deal with if they do their job.. even if you accidentally pick it up. **_uh oh_**.
> 
> if you're an antagonist that offs a contractor and wants to pinch their MOD for whatever reasoning, heck say your contractor master as a support died and you found their body with the tablet and they're unrevivable.. _I'm aware there's not frequent reasonings to do this as an antagonist but sometimes, that extra gear does, help._
> 
> as medical staffing, why would you revive the armored guy with an unremovable suit that batons people, bloodred grade armor, tons of contraband, probably freedom implants... much less if you lack thread rippers and they're just... **dead with total organ failure**, the suit covering their whole body and you want to revive them, ya can't.
> 
> and if you remove their uplink implant, the entire rest of the round is going to absolutely suck, for the player of the contractor.. so implant removal is just lame really to begin with especally if say the contractor had a massive streak of bad luck because you can only really either let them walk away with _all of their contraband_, or none of it. **yes, I don't mean the surgery, I mean using explosives to destroy everything on them, that includes implants like freedoms, EMPs, yep!**
> 
> _I'm aware cargo can circumvent this but its good to account for newbies that might not realise this, or busy medbay dealing with the 10 bodies of people that happened to die from say, a blob or toxins moment._
> 
> personal note: I know there are some security that do act in bad faith and do kill nearly all caught antagonists but... we really don't need to encourage this by things like this. most of the time its not even really sec killing antags, its crew.
> 
> much less a stab every single antag into crit mentality for low level threats or people with contractor batons and then leave them laying around the brig/morgue marked as DNR for the entire round for just existing because they're uncontainable.

I hear you. I reintroduced the bomb, with a twist now. It's an 8 second macrobomb, that is only armed IF the contractor locks the dna lock module. Same for ninja.

**Comment by StrangeWeirdKitten** [comment]
Make sure the main PR is edited to reflect the changes

**Comment by YakumoChen** [comment]
Macrobomb? Like the big boom?

Also are these roles made aware of this change or is it going to be something small and hard to notice in the modsuit menu

**Comment by radarseas** [comment]
personally... I'd just make **contractor batons DNA lock** on use and instead **shock and slightly damage someone that isn't its user**, this solves most of the balance issues at hand _and is ten times funnier_. and add the advanced DNA lock to the contract MOD (this module can't be removed by a non user and is EMPproof anyway. _**It also gibs people stupid enough to ignore all warnings of a DNA locked suit with it installed if they turn said suit on while wearing it.**_



contract MODs have a bluespace cell, you'll never ion it enough unless you spend 10 minutes shooting the same thing and recharging non-stop, MOD being backpacks are also acid proof anyway and if they were not you could just destroy someone's entire loudout, armor and equipment with a beaker of acid instantly with no counter than (don't get clicked or step into an indefinite puddle that someone keeps spamming with beakers).

a 5x5 explosion does not fix reasonings to stuff a contractor into evidence and never revive or interact with them with like 100 non-functional organ damage all over, it just makes it worse because the risk someone will ignore the warning and set it off anyway.


most of a contractor's syndicate equipment is in their chest rig anyway if they're worth their salt, which can be stripped anyway without risk and usually is, hell I've seen security not give two damns about wearing normal chest rigs considering the come from smuggler satchels too, their pinpointer will be and their tablet are usually in their pockets moreso because they don't want to lose their only means of progression as an antagonist to their MOD being destroyed. the only usual thing you get out of their modsuit contraband wise that isn't a module is a guarrenteed radio jammer or maybe poison kit/combat medkit, the latter of which cargo can entirely bypass with your 5000 credits per contract (saliyric/oxandrone pens/bloodloss pens).

they're nowhere near as much of lootbags as the syndicate supply pod that landed in the bar with codespeak and 3 cham kits plus a dsword or the one traitor that decided to not be boring and do traitor stuff, gimmic and got murdered and looted by 10 people. they don't even get starting TC anymore (which was a deserved change if their starting kits didn't range from useless (single magazine donksoft SMG with no spare ammo while you lack any and all TC) to somewhat okay (jaws of life that let you get that first contract and buy said donksoft SMG and extra mags plus riot darts off cargo and medipens _that work through your modsuit instantly and stack heal with omnizine, plus the modsuit armor, ah yes, thanks skyrat balance)._


also if you're a random antagonist that ran off with a contractor's body to revive it, you'll die from trying to get the MOD off because I guarrentee you won't and don't have nor the means to get a thread ripper and defib module to do the nessessary surgery. so this fucks over antags more, once again.



_tldr: skyrat balance 5000 credit into 20 medipens plus donksoft SMG with riots, plus free prybar jaws go brrr, being stunned by the baton than being stunned with the baton is funnier, 5x5 explosion, more like newbie bait, lootbox more like lootrig. their mod is just a backpack o' tiding tbh._

**Comment by zeroisthebiggay** [comment]
just played a round where after i died with my DNA lock activated and my MODsuit locked it didn't blow up when i got stripped
the MODsuit was disabled by my head being cut off but that shouldn't affect it right
do DNA locks get cleared if you turn the MOD on and off or something

**Comment by zeroisthebiggay** [comment]
round `2913` if you want to debug or something

**Comment by radarseas** [comment]
I just realised... aren't these basic DNA locks for contractors?... doesn't that mean a single EMP blast is gonna... wipe the DNA lock buffer and render the entire point of having it moot?

**Comment by radarseas** [comment]
> just played a round where after i died with my DNA lock activated and my MODsuit locked it didn't blow up when i got stripped the MODsuit was disabled by my head being cut off but that shouldn't affect it right do DNA locks get cleared if you turn the MOD on and off or something

I believe... its stripping that sets it off, not losing limbs, which means you're getting looted anyway if you get delimbed somehow.

**Comment by destrucktoid** [comment]
Right now i think this just needs to be closed and not pursued any further, this is for a variety of reasons:
1: This encourages security to just kill and round remove you as contractor/ninja as attempting to remove the suit COULD result in a macro-explosion
2: This encourages security to just throw the suits into space because they are bombs, thus making the modsuits irrecoverable should you manage to break out.
3: This, along with the previous "modsuits being un-removable" is completely a completely unnecessary change that doesn't change anything gameplay wise, all it does it adds additional hoops and makes things worse for both contractor and ninja players

The correct choice is what almost everyone has said from the beginning, **Just give them back the microbomb implant**. If you really object to the round removal from it then make the microbomb implant optional, IE it's in an implant injector inside of your modsuit and you need to inject it manually rather than jumping through all these weird hoops and making all these changes that aren't necessary and nobody who i have spoken to wants.

**Comment by Detective-Google** [comment]
doesn't really work, there are multiple ways to get the suit off and get the loot out of it without setting it off even when armed, even if security ignores it and pulls it off anyway the explosion doesn't do enough damage to properly incinerate the wearer's gear, nor even crit the people near them. If it doesn't destroy the gear, or blow a hole in the floor, or even kill anyone around or kill the wearer, what is the point of this as opposed to not existing at all.

**Comment by Detective-Google** [comment]
The microbomb is the correct solution here, it neatly removes the player and the loot pinata without causing any real extraneous damage, discouraging suicide bombing- you can be +carrying+ a contractor and have a microbomb go off and it barely injures you. If the  concern is the implant auto-detonating making it too "meta" to just kill them so they explode, just remove the automatic detonation and make it an action button that can be clicked while dead too.

**Comment by The-Sun-In-Splendour** [comment]
EMPs scramble the DNA lock _**even though the suit has EMP shield.**_ So you can literally just EMP the contractor once and steal his modsuit, it says the hugboxed warning message "hooo a failsafe bomb!!!" but it never arms. Free contractor modsuit!

File: modular_zubbers/code/modules/antagonists/modglue.dm
Comment (The-Sun-In-Splendour) [inline modular_zubbers/code/modules/antagonists/modglue.dm:59]: ???? Why are you doing this??? I thought the whole point was them not being loot pinatas??? I legitimately do not understand where you're going with this PR. Why are you making the DNA lock EMP resistant _**JUST**_ to make it so pulling out the suit dumps it's contents???

Comment (The-Sun-In-Splendour) [reply modular_zubbers/code/modules/antagonists/modglue.dm:59]: Just give them a goddamn microbomb at this point.

Comment (nevimer) [reply modular_zubbers/code/modules/antagonists/modglue.dm:59]: I don't think you understand the point. Based on feedback, players wanted to be able to get stolen things back from contractors.  They wanted to be able to retreive AI's and gear - This doesn't dump the modules. They also didn't like the macrobomb, so now we get a microbomb that blows up the suit (after dumping its contents).

Comment (StrangeWeirdKitten) [reply modular_zubbers/code/modules/antagonists/modglue.dm:59]: There will be a dump contents added to empty the modsuit storage as per my requirement for this PR.

Comment (Detective-Google) [reply modular_zubbers/code/modules/antagonists/modglue.dm:59]: > There will be a dump contents added to empty the modsuit storage as per my requirement for this PR.

this defeats the entire point of having the suit self destruct- it doesn't actually stop them being a loot pinata in anyway except removing the baton from the round

Comment (Detective-Google) [reply modular_zubbers/code/modules/antagonists/modglue.dm:59]: I'd also say that retrieving stolen items is a big reason not to instantly kill contractors- why not just shoot them to take the items off their corpse if you have 0 reason to take them in alive. There are very few items that are truly irreplacable and almost none of them are round critical, there's the very niche cases of the contractor lugging brains or AI cards in their bag but as per the RR and the nature of their objectives there's very little reason they'd do either.-- To put it simpler, if the contractor happens to steal the X0-1 earlier in the round it gives the HoS an incentive to detain them with non lethals/not let them die whereas with the current system there's no reason not to just shoot them to death on the spot and strip their suit for the free antag goodies---- **INCLUDING A PINPOINTER THAT IGNORES SUIT SENSORS**

**Comment by Detective-Google** [comment]
https://cdn.discordapp.com/attachments/700317966943649817/1186553785519177744/Untitled_video_-_Made_with_Clipchamp_18.mp4?ex=6593ab53&is=65813653&hm=afcf22c5c79a48976d7e44cb505c1a76daa2b91823054cf25ef389698ec00dec& for an example of why this doesn't work, RID 3075, drops any loot the contractor might have on them completely negating having a bomb at all. Gives the station sec access to a pinpointer that works on anyone with an ID, + syndie comm, + whatever junk they happened to buy beforehand.

**Comment by StrangeWeirdKitten** [comment]
I plan on moving contractor to the TG traitor kits anyway. Right now I mostly care about Ninja with this PR.

**Comment by Detective-Google** [comment]
It could also be made totally non kosher from an administrative standpoint for security to use antag gear (save for like delta ig) but currently since I often see security running around with looted traitor gear, eswords, et cetera it definitely needs to be a large consideration. Were it the case they just weren't allowed to touch I wouldn't have to care about Security getting this stuff from dead tractors.

Nevimer said something interesting earlier when I was combing trying to figure out +why+ people were so opposed to the microbomb- in that he said he believed in the premise of the original pr removing them, the body of which, (paraphrasing;) reads as follows: 

"If antagonists can instantly delete themselves when they're not winning it encourages them to only play when they're winning."

I think this is a pretty fair observation, but I'd also like to interrogate it a little.
I don't think most antagonists would choose to delete themselves if they weren't winning but **were** having fun- essentially, if an antagonist can immediately remove themselves when they're _no longer having fun_, they're going to do so. I think we should consider what the factors are that make it so that losing is so un-fun for players that this is the case.

- Bad player culture, simple as, the biggest reason is that as soon as security drops you they're going to parade your body around, gloat, and brag, and generally display poor sportsmanship. Part of this comes down to reclaiming the lost dignity of losing earlier fights("I FINALLY got this fucker after he killed me three times"), part of it is just the adrenaline rush of winning a fight. I can't honestly say I don't feel the urge to pump my hands and cheer (metaphorically) after a hard won fight, but I also understand that you should respect the other people trying to play the game and be thankful they gave you a good challenge, so I temper that feeling and don't project it onto others to make them feel shitty.

- Dropping a bunch of loot, This is a more secondary concern, but the current server policy is rather cavalier about antag loot being used, and I've seen on multiple occasions recently security breaking eswords or other very strong antagonist gear out of evidence to go hunt down other antags-- or just taking it off their body and holding onto it. The game as we've chosen to play it here doesn't really support this and it quickly means security is going to roll over any antagonist, which kills the round pacing and makes for a bit of a snoozefest. 

- Your round is effectively over when you lose. Almost all security officers having defeated an antagonist will simply dump them in perma and forget about them-- and this isn't to say this is a failing on them, not everyone wants to deal with long prisoner scenes, and some people are selective about who they want to engage in that kind of stuff with, they only want to spend their time interrogating someone who's going to provide a good interrogation, essentially. I also really hate to have space law become so weak so that murderers are just a slap on the wrist and let out of a cell either- as it was on skyrat for a time, because this makes security incredibly frustrating when players can just do their objectives, stash their illicit items, and then sit in the brig for full amnesty. But however you shake it the fact that your round is confined to either sitting around in perma watching paint dry, or just attempting to beat up any officer who enters(which means they'll just kill you entirely instead-- justifiably), what's the point in staying in the round once you've lost for them? other than to be a walking trophy for the security team. This is, of course, assuming security bothers to have the contractor treated in the first place, which is also pretty uncommon. 

my TLDR is that these prs are addressing symptoms, not the causes, of server culture and policy coming together to create scenarios where if you're not winning there's hardly any point to play.

To fellow players, remember, respect your antagonist players, enjoy the good fights with them. It's a lot better than having people be post-mortem bullied for daring to lose and turning off antagonists entirely.

**Comment by destrucktoid** [comment]
> It could also be made totally non kosher from an administrative standpoint for security to use antag gear (save for like delta ig) but currently since I often see security running around with looted traitor gear, eswords, et cetera it definitely needs to be a large consideration. Were it the case they just weren't allowed to touch I wouldn't have to care about Security getting this stuff from dead tractors.
> 
> Nevimer said something interesting earlier when I was combing trying to figure out +why+ people were so opposed to the microbomb- in that he said he believed in the premise of the original pr removing them, the body of which, (paraphrasing;) reads as follows:
> 
> "If antagonists can instantly delete themselves when they're not winning it encourages them to only play when they're winning."
> 
> I think this is a pretty fair observation, but I'd also like to interrogate it a little. I don't think most antagonists would choose to delete themselves if they weren't winning but **were** having fun- essentially, if an antagonist can immediately remove themselves when they're _no longer having fun_, they're going to do so. I think we should consider what the factors are that make it so that losing is so un-fun for players that this is the case.
> 
> * Bad player culture, simple as, the biggest reason is that as soon as security drops you they're going to parade your body around, gloat, and brag, and generally display poor sportsmanship. Part of this comes down to reclaiming the lost dignity of losing earlier fights("I FINALLY got this fucker after he killed me three times"), part of it is just the adrenaline rush of winning a fight. I can't honestly say I don't feel the urge to pump my hands and cheer (metaphorically) after a hard won fight, but I also understand that you should respect the other people trying to play the game and be thankful they gave you a good challenge, so I temper that feeling and don't project it onto others to make them feel shitty.
> * Dropping a bunch of loot, This is a more secondary concern, but the current server policy is rather cavalier about antag loot being used, and I've seen on multiple occasions recently security breaking eswords or other very strong antagonist gear out of evidence to go hunt down other antags-- or just taking it off their body and holding onto it. The game as we've chosen to play it here doesn't really support this and it quickly means security is going to roll over any antagonist, which kills the round pacing and makes for a bit of a snoozefest.
> * Your round is effectively over when you lose. Almost all security officers having defeated an antagonist will simply dump them in perma and forget about them-- and this isn't to say this is a failing on them, not everyone wants to deal with long prisoner scenes, and some people are selective about who they want to engage in that kind of stuff with, they only want to spend their time interrogating someone who's going to provide a good interrogation, essentially. I also really hate to have space law become so weak so that murderers are just a slap on the wrist and let out of a cell either- as it was on skyrat for a time, because this makes security incredibly frustrating when players can just do their objectives, stash their illicit items, and then sit in the brig for full amnesty. But however you shake it the fact that your round is confined to either sitting around in perma watching paint dry, or just attempting to beat up any officer who enters(which means they'll just kill you entirely instead-- justifiably), what's the point in staying in the round once you've lost for them? other than to be a walking trophy for the security team. This is, of course, assuming security bothers to have the contractor treated in the first place, which is also pretty uncommon.
> 
> my TLDR is that these prs are addressing symptoms, not the causes, of server culture and policy coming together to create scenarios where if you're not winning there's hardly any point to play.
> 
> To fellow players, remember, respect your antagonist players, enjoy the good fights with them. It's a lot better than having people be post-mortem bullied for daring to lose and turning off antagonists entirely.

Very much agree with the sentiment here. If i'm a contractor and i lose and someone i know is on security then more often than not i'll stay in round and engage with them. Otherwise if i lose i'd honestly just prefer you keep me dead because i'd rather be a ghost and watch other people play, possibly roll another midround or go to play another game than just sit in perma bored out of my brains.

regarding your point on player culture, i agree sometimes and disagree others, it really depends upon who is on as security in the shift as to how bad this is or how good it is. For example there have been times when i'll see security go absolutely ham and validhunt you to death just for existing as a traitor, constantly trying to do what i call antag bait you (keep trying to bait you into fighting them in order to force conflict rather than being willing to go along with a more peaceful gimmick of some kind.). Then there are other shifts where security is perfectly fine and are perfectly content to go along with antags and be somewhat friendly. It entirely depends upon who is playing security that shift.

Regarding antag loot, i would like to remind you that according to space law antag gear should be stored in contraband and shouldn't be used by security (outside of red alert). I will admit though that there are exceptions to this rule and people do ignore it and get away with it.

**Comment by Detective-Google** [comment]
Unless it becomes against the rules OOCly to do it as well people won't care, IC repercussions are not enough to dissuade the allure of "ooh loot"

and yeah, obv this depends on who's playing heavily, the *average* sec is like this, there are of course standouts but I find them less common these days. 

There's also the matter that since you can't get starting TC anymore it's very hard to stealth since you're generally going to have to pull up on someone with the baton and no fun disguise for your first contract; at which point people are more than allowed to light you up.

It honestly feels like with this and the TC removal contractors are just targets for security to go and shoot and then parade around the bodies as trophies.

**Comment by The-Sun-In-Splendour** [comment]
> I plan on moving contractor to the TG traitor kits anyway. Right now I mostly care about Ninja with this PR.

Then this PR is literally a nothingburger. Ninjas already by design cannot be walking lootbags because their modsuit _**GIBS YOU**_ if you put it on without the correct DNA and ID. In fact you're making it all worse because now if a ninja dies they're going to dump all their items on the floor.

**Comment by Screampuff** [comment]
This is an absolutely dreadful PR that should be withdrawn from testmerge.

Contractors can't be imprisoned because their gear cannot be removed without exploding or going through a process to EMP them with kid gloves, ergo the only way to handle a contractor is to round remove them.  Nobody wins.  This doesn't encourage interaction or roleplaying and it does nothing to further gameplay, especially when the alternative is literally to just *DNA lock the suit and baton* if that's the major concern.  Making the contractor into a maxcap is a ridiculous way to keep gamer gear out of the hands of crew.

**Pull request merged**
